src/main/java/com/klarna/hiverunner/CommandShellEmulation.java

```diff
@@ -35,6 +53,25 @@ public String transformScript(String script) {
     }
   },
   BEELINE {
+
+    private final String runCommand = "!run";
+
+    @Override
+    public boolean isImportFileStatement(String statement) {
+      // case-sensitive
+      return statement.trim().startsWith(runCommand);
+    }
+
+    @Override
+    public File getImportFileFromStatement(String statement) {
+      // filename cannot contain whitespace
+      String[] tokens = statement.trim().split(" ");
+      if (tokens.length == 2) {
+        return new File(tokens[1]);
+      }
+      throw new IllegalArgumentException("Cannot get file to import from '" + statement + "'");
+    }
+
     @Override
     public String transformStatement(String statement) {
       return filterFullLineComments(statement);
```

Comment:
The implementation seems to only support `!<command>` (see [BeeLine#dispatch](https://github.com/apache/hive/blob/master/beeline/src/java/org/apache/hive/beeline/BeeLine.java#L1128)).